wallet-new/src/Cardano/Wallet/API/V1/Handlers/Wallets.hs

```diff
@@ -97,3 +102,20 @@ updateWallet pwl wid walletUpdateRequest = do
     case res of
          Left e  -> throwM e
          Right w -> return $ single w
+
+getUtxoStatistics :: PassiveWalletLayer IO
+                  -> WalletId
+                  -> Handler (WalletResponse UtxoStatistics)
+getUtxoStatistics pwl wid = do
+    res <- liftIO $ WalletLayer.getUtxos pwl wid
+    case res of
+         Left e  -> throwM e
+         Right w ->
+             let
+                 extractValue :: TxOutAux ->  Word64
+                 extractValue = getCoin . txOutValue . toaOut
+                 utxosCoinValuesForAllAccounts :: [(Account, Utxo)] -> [Word64]
+                 utxosCoinValuesForAllAccounts pairs =
+                     concatMap (\pair -> map extractValue (M.elems $ snd pair) ) pairs
+             in do
+                 return $ single (computeUtxoStatistics $ utxosCoinValuesForAllAccounts w)
```

Comment:
This reads better with the `do` at the upper-most level:

```hs
         Right w -> do
             let extractValue :: TxOutAux ->  Word64
                 extractValue = 
                     getCoin . txOutValue . toaOut
                 
             let utxosCoinValuesForAllAccounts :: [(Account, Utxo)] -> [Word64]
                 utxosCoinValuesForAllAccounts pairs =
                     concatMap (\pair -> map extractValue (M.elems $ snd pair) ) pairs
              
             return $ single (computeUtxoStatistics $ utxosCoinValuesForAllAccounts w)
```

Also, I'd rather move these functions inside the `computeUtxoStatistics` function. It will also make its signature a bit clearer. Rather than `[Word64] ->  UtxoStatistics` it becomes `NonEmpty (Account, Utxo) -> UtxoStatistics` which in my opinion gives already a much clearer idea of what the function is doing and how to use it.

Reply:
applied within `do` . As underscored above I moved everything to `Cardano/Wallet/Types/UtxoStatistics.hs` which is reexported in  V1/Types.hs. I retained `lets` here as there is Account which means in order to avoid cyclic dependencies one solution would be to move everything related with `Account` to separate `Cardano/Wallet/Types/Account.hs` and reexport it in V1.Types.hs. I am not sure if we want this